Core/OfficeDevPnP.Core/Framework/Provisioning/ObjectHandlers/ObjectNavigation.cs

```diff
@@ -185,6 +180,48 @@ public override TokenParser ProvisionObjects(Web web, ProvisioningTemplate templ
 
         #region Utility methods
 
+        private bool WebSupportsProvisionNavigation(Web web, ProvisioningTemplate template)
+        {
+            if (template.Navigation == null) return true;
+
+            bool isNavSupported = true;
+            // The Navigation handler for managed metedata only works for sites with Publishing Features enabled
+            if (!web.IsPublishingWeb())
+            {
+                if (template.Navigation.GlobalNavigation != null
+                    && template.Navigation.GlobalNavigation.NavigationType == GlobalNavigationType.Managed)
+                {
```

Comment:
Hi again,
Moreover here I think that if the template has for example managed navigation for GlobalNavigation and structural navigation for CurrentNavigation, which is uncommon but possible, the code will somehow fail. I mean, we would fully skip the provisioning or the extraction (in lines from 213 to 220) even if theoretically we should provision or extract one of the two. Have you tested this scenario? Thanks.